etc/testing/circle/workloads/go-test-results/collector/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pachyderm/pachyderm/v2/src/client"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	projectName = "ci-metrics"
+	repoName    = "go-test-results-raw"
+	fileSuffix  = "go-test-results.jsonl"
+)
+
+var pachdAddress = findPachdAddress()
+var logger = log.Default()
+
+func findPachdAddress() string {
+	env := os.Getenv("OPS_PACHD_ADDRESS")
+	if len(env) > 0 {
+		return env
+	}
+	return "grpcs://pachyderm.pachops.com:443"
+}
+
+func sanitizeName(s string) string {
+	ret := []byte(strings.ReplaceAll(s, " ", "_"))
+	invalidCharacters := regexp.MustCompile("[^a-zA-Z0-9-_]+")
+	ret = invalidCharacters.ReplaceAll(ret, []byte("-"))
+	return string(ret)
+}
+
+func findDestinationPath(path string, branch string, jobName string, resultsFolder string) string {
+	tmpPath := strings.Split(strings.TrimPrefix(path, resultsFolder), "/") // remove standard /tmp/test-results/ folder
+	tmpPath = append([]string{branch, jobName}, tmpPath...)                // set up output file location
+	return filepath.Join(tmpPath...)
+}
+
+func main() {
+	logger.Println("Beginning results upload.")
+	robotToken := os.Getenv("PACHOPS_PACHYDERM_ROBOT_TOKEN")
+	if len(robotToken) == 0 {
+		logger.Println("No pachyderm robot token found. Continuing with unauthenticated pach client.")
+	}
+	resultsFolder := os.Getenv("TEST_RESULTS")
+	if len(resultsFolder) == 0 {
+		logger.Fatal("TEST_RESULTS needs to be populated to find the test results folder.")
+	}
+	branch := sanitizeName(os.Getenv("CIRCLE_BRANCH"))
```

Comment:
@djanicekpach INT i believe also run on releases. in that case there will be no CIRCLE_BRANCH var instead you will see a CIRCLE_TAG. you should not fail when CIRCLE_BRANCH is null. Will be also nice to query later across tag of the release.

Reply:
Added handling for this